docs/v1beta1-checklist-developer-guide.md

```diff
@@ -0,0 +1,126 @@
+---
+title: v1beta1 Checklist
+toc: true
+weight: 740
+indent: true
+---
+# v1beta1 Checklist
+
+Crossplane adheres to the [Kubernetes definition] of API versioning. This means
+that when an API type reaches a `beta` level of quality it should be
+well-tested, enabled by default, and will be supported through subsequent
+releases, even if the details of the type change. As a community, we implement
+`beta` API types when the following are true:
+
+- An `alpha` version of the API types currently exits.
+- We are committed to long-term support the API type.
+- There is demonstrated interest for support of the API type in the end-user
+  community.
+
+When the above are all true, the first version of a `beta` level API type,
+`v1beta1`, should be implemented in accordance with the specifications described
+in the remainder of this document.
+
+## Areas of Quality
+
+Crossplane controllers are comprised of four main components: the cloud provider
+API type, the Crossplane API type, the client, and the controller. Each of these
+components serves a specific purpose that should not overlap with any of the
+other components. It is helpful to conceptualize these components in the
+following manner:
+
+- **Cloud Provider API Type**: the language the controller speaks to manage the
+  cloud provider resource
+- **Crossplane API Type**: the language the controller speaks to manage the
+  Crossplane resource (Kubernetes [CustomResource])
+- **Client**: the translator between the Crossplane API type and the cloud
```

Comment:
It could make sense to denote that `Client` is a set of functions and then describe it as `translator`.

Reply:
Added to section below per my comment above.